src/compute/src/render/mod.rs

```diff
@@ -125,6 +126,7 @@ use mz_storage::boundary::ComputeReplay;
 pub mod context;
 mod flat_map;
 mod join;
+mod persist_source;
```

Comment:
Nit, but the intended separation of responsibilities would probably put this in STORAGE, like the other `ComputeReplay` implementors. Like, if STORAGE wants to change how the implementation of this source works, they should be able to and COMPUTE should just have to link against their code.

Reply:
That's a good point! I'll move the code to the `storage` crate. It would also be nice to "hide" it inside the boundary but that would require implementing code in all the `ComputeReplay` impls (of which there is only one real one) because the boundary/replay impls are orthogonal to replay via persist. Plus, in the long run the current `ComputeReplay` impls will go away and we will have only persist replay.

I can't just add a `PersistReplay` right now because the boundary is not instantiated per source but is global to the compute instance. Is that what you had in mind?